Slightly reduce the Depth3Test that often times out on Linux

trylek opened the pull request:
This is one of the regression tests I introduced as part of implementing protection against infinite generic recursion in Crossgen2. Turns out that on the relatively slow Helix Linux VMs the test often times out due to compilation time exceeding 10 minutes. I propose slightly trimming down the test so that it uses less time in Helix.

Thanks

Tomas

/cc @dotnet/crossgen-contrib, @dotnet/runtime-infrastructure

trylek committed:
Slightly reduce the Depth3Test that often times out on Linux

This is one of the regression tests I introduced as part of
implementing protection against infinite generic recursion in
Crossgen2. Turns out that on the relatively slow Helix Linux
VMs the test often times out due to compilation time exceeding
10 minutes. I propose slightly trimming down the test so that
it uses less time in Helix.

Thanks

Tomas

ivdiazsa commented:
Do we know if this is a natural problem with not-too-fast machines or whether we have an unnatural slowdown we ought to investigate and fix?

trylek commented:
Well, in https://github.com/dotnet/runtime/pull/87532 I proposed a minor optimization in the algorithm but the PR was abandoned as Michal pointed out that this is more or less a pathological case we don't need to optimize for. In a similar vein, on my quite fast devbox the test in checked mode takes about 1 minute, I can easily imagine it can be 10 times slower on a shared Azure VM.

ivdiazsa commented:
> Well, in #87532 I proposed a minor optimization in the algorithm but the PR was abandoned as Michal pointed out that this is more or less a pathological case we don't need to optimize for. In a similar vein, on my quite fast devbox the test in checked mode takes about 1 minute, I can easily imagine it can be 10 times slower on a shared Azure VM.

I see. Well, in that case it LGTM! I've restarted the failing pipeline because it seems entirely unrelated to this change. So unless @MichalStrehovsky has further feedback, I think we can merge this, assuming that failure is indeed unrelated like I suspect it to be.

trylek committed:
Remove issues.targets exclusion for the Depth3Test

trylek commented:
Thanks Andy and Ivan for reviewing. I have rebased the change against latest main and I have added the removal of the clause for ```Depth3Test``` from issues.targets where Bruce added it last week to baseline the relatively frequent lab failure. I don't think it's necessary to port this basically negative test to .NET 8 RC1 - let me know if you think otherwise.

ivdiazsa commented:
Wasm failure with Mono is a known infrastructure issue unrelated to this change, so I'll be merging this right now.
